src/test/java/utils/GeneralConfigModule.java

```diff
@@ -21,15 +21,28 @@ public void applyToConfigBuilder(SchemaGeneratorConfigBuilder builder) {
 		builder.forTypesInGeneral()
 				.withSubtypeResolver(new ClassGraphSubtypeResolver())
 				.withCustomDefinitionProvider(descriptionProvider)
-				.withTypeAttributeOverride(descriptionProvider);
+				.withTypeAttributeOverride(descriptionProvider)
+				//.withCustomDefinitionProvider(new ObjectIdToCustomSchemaProvider())
+		;
+		builder.forFields()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides);
 		builder.forMethods()
+				.withTargetTypeOverridesResolver(this::collectTargetTypeOverrides)
 				.withIgnoreCheck(method -> method.getSchemaPropertyName().endsWith(")"));
-		builder
-//				.with(new SimpleTypeModule().withStringType(ObjectId.class))
-				.forTypesInGeneral()
-				.withCustomDefinitionProvider(new ObjectIdToCustomSchemaProvider())
-		;
+	}
 
+	private List<ResolvedType> collectTargetTypeOverrides(MemberScope<?, ?> member) {
+		if (member.getType() == null || !member.getType().isInstanceOf(ObjectId.class)) {
```

Comment:
The `null` check is technically only required for `void` methods, but it doesn't hurt I guess. 😉